cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -102,6 +105,10 @@ public class KafkaAssemblyOperator extends AbstractAssemblyOperator<KubernetesCl
         OPERATOR_VERSION = PROPERTIES.getProperty("version");
     }
 
+    /**
+     * Intentionally shadowing inherited field to make it in specific implementation dedicated for the Kafka assembly
+     */
```

Comment:
Why do we need to shadow the field? Doesn't it extend the MetricsHolder?

Reply:
Because we need the `KafkaAssemblyOperatorMetricsHolder` instance to access `removeMetricsForCertificates`, `clusterCaCertificateExpiration` and `clientCaCertificateExpiration`. Yes we can use the `MetricsHolder` instance, but then we need to typecast it every time we need it. IMO not a fan of typecasting.